python_ta/__init__.py

```diff
@@ -117,42 +80,87 @@ def _check(module_name='', reporter=ColorReporter, number_of_messages=5, level='
     try:
         # Check for inline "pylint:" comment, which may indicate a student
         # trying to disable a check.
-        # TODO: Put this into a helper function.
-        with tokenize.open(spec.origin) as f:
+        with tokenize.open(spec) as f:
             for (tok_type, content, _, _, _) in tokenize.generate_tokens(f.readline):
                 if tok_type != tokenize.COMMENT:
                     continue
                 match = pylint.utils.OPTION_RGX.search(content)
                 if match is None:
                     continue
                 else:
-                    print('ERROR: string "pylint:" found in comment. No checks will be run.')
-                    return
+                    print('ERROR: string "pylint:" found in comment. ' +
+                          'No checks will be run.')
+                    return False
     except IndentationError as e:
         print('ERROR: python_ta could not check your code due to an ' +
               'indentation error at line {}'.format(e.lineno))
-        return
+        return False
     except tokenize.TokenError as e:
         print('ERROR: python_ta could not check your code due to a ' +
               'syntax error in your file')
+        return False
+    return True
+
+
+def _check(module_name='', reporter=ColorReporter, number_of_messages=5, level='all',
+           local_config_file='', pep8=False):
+    """Check a module for problems, printing a report.
+
+    <level> is used to specify which checks should be made.
+
+    The name of the module should be the name of a module,
+    or path(s) to Python file(s) or directory(ies).
+    """
+    # Check if `module_name` is not the type str, raise error.
+    if not isinstance(module_name, list):
+        print('No checks run. Input to check, `{}`, has invalid type, must be a list of strings.'.format(module_name))
         return
+    for item in module_name:
+        if not isinstance(item, str):
+            print('No checks run. Part of input to check, `{}`, has invalid type, must be type: str.'.format(item))
+            return
 
+    current_reporter = reporter(number_of_messages)
+    linter = _load_pylint_plugins(current_reporter, local_config_file, pep8)
+    patch_all()  # Monkeypatch pylint
     try:
-        linter.check([spec.origin])
-        current_reporter.print_messages(level)
+        linter.open()  # initialize stats
+
+        # print(' '.join(module_name))
+
+        expanded_files, errors = utils.expand_modules(module_name, 
+                        linter.config.black_list, linter.config.black_list_re)
+        for error in errors:
+            current_reporter.show_file_linted(error['mod'])
+            if isinstance(error['ex'], ImportError):
+                print('Error: {}, which means the file "{}" could not be found.\n'
+                      .format(error['ex'], error['mod']))
+            else:
+                print('Error: {}\n'.format(error['ex']))
+        print()
```

Comment:
This call to `print` isn't very elegant. You should combine this with a previous call, or a future one.